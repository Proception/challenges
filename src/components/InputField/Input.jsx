import React from 'react';
import PropTypes from 'prop-types';

const Input = (props) => {
  return (
    <input 
      className={props.className}
      type={props.type}
      min={props.min}
      name= {props.name}
      value={props.value} 
      onClick={props.action}
      onChange={props.handleChange}
      placeholder={props.placeholder}
    />
  );
};

Input.propTypes = {
  min: PropTypes.number,
  className: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  placeholder: PropTypes.string
};

export default Input;
