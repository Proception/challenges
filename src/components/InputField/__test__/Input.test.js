import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input.jsx';


test('Input component', () => {
  const component = renderer.create(
    <Input />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
