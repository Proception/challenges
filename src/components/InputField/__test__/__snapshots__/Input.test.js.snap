// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component 1`] = `<input />`;
